app/src/main/java/com/hedvig/app/feature/marketing/MarketingViewModel.kt

```diff
@@ -167,37 +119,17 @@ class MarketingViewModel(
   fun onClickLogIn() {
     hAnalytics.buttonClickMarketingLogin()
   }
-
-  companion object {
-    private val PICK_MARKET_INITIAL = PickMarket(
-      isLoading = true,
-      isValid = false,
-      market = null,
-      language = null,
-      availableMarkets = emptyList(),
-    )
-  }
 }
 
-sealed interface ViewState
-
-object Loading : ViewState
-
-data class PickMarket(
-  val isLoading: Boolean,
-  val isValid: Boolean,
-  val market: Market?,
-  val language: Language?,
-  val availableMarkets: List<Market>,
-) : ViewState
-
-sealed interface MarketPicked : ViewState {
-  object Loading : MarketPicked
-
-  data class Loaded(
-    val selectedMarket: Market,
-    val loginMethod: LoginMethod,
-  ) : MarketPicked
+data class ViewState(
+  val market: Market? = null,
+  val language: Language? = null,
+  val availableMarkets: List<Market> = emptyList(),
+  val selectedMarket: Market? = null,
+  val loginMethod: LoginMethod? = null,
+  val isLoading: Boolean = true,
+) {
+  fun canSetMarketAndLanguage() = market != null && language != null
```

Comment:
I love the win we get here with this basically being a computed property instead of manually having to check for it when changing the object. Super nice!